examples/Basic/gui_estimator.py

```diff
@@ -0,0 +1,135 @@
+
+    def execute(self):
+        log.info("Starting to generate numbers")
+        for i in range(self.iterations):
+            data = {
+                'Iteration': i,
+                'Random Number': random.random()
+            }
+            log.debug("Produced numbers: %s" % data)
+            self.emit('results', data)
+            self.emit('progress', 100*i/self.iterations)
+            sleep(self.delay)
+            if self.should_stop():
+                log.warning("Catch stop command in procedure")
+                break
+
+    def get_estimates(self, sequence_length=None, sequence=None):
+        """ Function that returns estimates for the EstimatorWidget. If this function
+        is implemented (and does not return a NotImplementedError) the widget is
+        automatically activated.
+
+        The function is expected to an int or float, or a list of tuples. If an int or
+        float is returned, it should represent the duration in seconds.If a list of
+        tuples is returned, each tuple containing two strings, a label and the estimate
+        itself:
+        estimates = [
+            ("label 1", "estimate 1"),
+            ("label 2", "estimate 2"),
+        ]
+        The length of the number of estimates is not limited but has to remain unchanged after
+        initialisation. Not that also the label can be altered after initialisation.
+
+        The keyword arguments `sequence_length` and `sequence` are optional and return
+        (if asked for) the length of the current sequence (of the `SequencerWidget`) or
+        the full sequence.
+
+        """
+        duration = self.iterations * self.delay
+
+        """
+        A simple implementation of the get_estimates function immediately returns the duration
+        in seconds.
+        """
+        # return duration
+
+        estimates = list()
+
+        estimates.append(("Duration", "%d s" % int(duration)))
+        estimates.append(("Number of lines", "%d" % int(self.iterations)))
+
+        estimates.append(("Sequence length", str(sequence_length)))
+
+        estimates.append(('Measurement finished at', str(datetime.now() + timedelta(
+            seconds=duration))[:-7]))
+        estimates.append(('Sequence finished at', str(datetime.now() + timedelta(
+            seconds=duration * sequence_length))[:-7]))
+
+        return estimates
+
+    def shutdown(self):
+        log.info("Finished")
+
+
+class MainWindow(ManagedWindow):
+
+    def __init__(self):
+        super(MainWindow, self).__init__(
+            procedure_class=TestProcedure,
+            inputs=['iterations', 'delay', 'seed'],
+            displays=['iterations', 'delay', 'seed'],
+            x_axis='Iteration',
+            y_axis='Random Number',
+            sequencer=True,
+            sequence_file="gui_sequencer_example_sequence.txt"
+        )
+        self.setWindowTitle('GUI Example')
+
+    def queue(self, procedure=None):
+        filename = tempfile.mktemp()
```

Comment:
`def queue(self, procedure=None):` should be replaced with `def queue(self, *, procedure=None):` as in the `gui_sequencer.py` example, otherwise when clicking to "Queue" button it will fail. Reason for this is that the method `queue` is always called with a single parameter with value `False` when button "Queue" is clicked (`checked` parameter in the `QAbstractButton` `clicked` signal) and with a keyword parameter `procedure` when "Queue sequence" button is clicked. Possibly this mechanism needs to be updated in the future as it is error-prone.

Reply:
Ahh yeah, that's a good point indeed

Reply:
I meant that the `queue` method is defined in the code and documentation without parameters, while it is called with a positional parameter set to `False` when activated by "Queue" button click and with a `procedure` keyword parameter when called by a sequencer.
So, if you initially write `queue` without parameters and later you add a sequencer, code will not work. On the other hand, if you add the `procedure` parameter, then you have to use the '*' syntax to avoid the positional parameter.  A bit weird, in my opinion.

Reply:
Ah I see what you mean. A possible would be to have a very simple wrapper which gets called by the queue button, which simply calls the queue method without any positional arguments (i.e. the False) to fix that issue.
For the sequencer, the only thing I could think of at this moment is to insert an inspection of the queue method upon initializing the sequencer and raise an error if the method doesn't accept the keyword argument. We could also consider adding the procedure keyword argument to the default implementation; in the basic gui example is also possible, though I'm a bit reluctant to add it there as well because I think it'd be best to keep that example as basic as possible.

What do you think? Or do m you have any other suggestions?

Reply:
Using the simple wrapper to discard the unwanted parameter is, in my opinion, the best approach.

Reply:
Implemented both the wrapper and the signature check